File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int divisor) {
        int[] answer = {};
        
        List<Integer> ansList = new ArrayList<>();
        
        for(int num : arr){
            if(num % divisor == 0){
                ansList.add(num);
            }
        }
        if(ansList.size() == 0){
            return new int[]{-1};
        }
        
        Collections.sort(ansList);
        
        answer = new int[ansList.size()];
        for(int i = 0; i < ansList.size(); i++){
            answer[i] = ansList.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.js
function solution(arr, divisor) {
    let answer = [];
    //for (a of arr) if (a % divisor === 0) answer.push(a);
    answer = arr.filter(value => value % divisor === 0)
    // if (answer.length === 0) return [-1];
    return answer.length === 0 ? [-1] : answer.sort((a, b) => a - b);
}